catalogd/cmd/catalogd/main.go

```diff
@@ -74,144 +73,232 @@ const (
-		webhookPort          int
-		caCertDir            string
-		globalPullSecret     string
-	)
-	flag.StringVar(&metricsAddr, "metrics-bind-address", "", "The address for the metrics endpoint. Requires tls-cert and tls-key. (Default: ':7443')")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.StringVar(&pprofAddr, "pprof-bind-address", "0", "The address the pprof endpoint binds to. an empty string or 0 disables pprof")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
-	flag.StringVar(&systemNamespace, "system-namespace", "", "The namespace catalogd uses for internal state, configuration, and workloads")
-	flag.StringVar(&catalogServerAddr, "catalogs-server-addr", ":8443", "The address where the unpacked catalogs' content will be accessible")
-	flag.StringVar(&externalAddr, "external-address", "catalogd-service.olmv1-system.svc", "The external address at which the http(s) server is reachable.")
-	flag.StringVar(&cacheDir, "cache-dir", "/var/cache/", "The directory in the filesystem that catalogd will use for file based caching")
-	flag.BoolVar(&catalogdVersion, "version", false, "print the catalogd version and exit")
-	flag.DurationVar(&gcInterval, "gc-interval", 12*time.Hour, "interval in which garbage collection should be run against the catalog content cache")
-	flag.StringVar(&certFile, "tls-cert", "", "The certificate file used for serving catalog and metrics. Required to enable the metrics server. Requires tls-key.")
-	flag.StringVar(&keyFile, "tls-key", "", "The key file used for serving catalog contents and metrics. Required to enable the metrics server. Requires tls-cert.")
-	flag.IntVar(&webhookPort, "webhook-server-port", 9443, "The port that the mutating webhook server serves at.")
-	flag.StringVar(&caCertDir, "ca-certs-dir", "", "The directory of CA certificate to use for verifying HTTPS connections to image registries.")
-	flag.StringVar(&globalPullSecret, "global-pull-secret", "", "The <namespace>/<name> of the global pull secret that is going to be used to pull bundle images.")
-
-	klog.InitFlags(flag.CommandLine)
-
-	// Combine both flagsets and parse them
-	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
-	features.CatalogdFeatureGate.AddFlag(pflag.CommandLine)
-	pflag.Parse()
-
-	if catalogdVersion {
-		fmt.Printf("%#v\n", version.Version())
-		os.Exit(0)
+	cfg := &config{}
+	cmd := newRootCmd(cfg)
+	if err := cmd.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func newRootCmd(cfg *config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "catalogd",
+		Short: "Catalogd is a Kubernetes operator for managing operator catalogs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(cfg)
+		},
 	}
 
+	flags := cmd.PersistentFlags()
```

Comment:
Persistent flags apply to all subcommands of this command. Since this command doesn't have any subcommands, I think it would make more sense to use `cmd.Flags()` here.